Medieval Content Mod Series 1

chaosvolt opened the pull request:
Weapons, bronze, and the armament of imperial China.

This is the first part of a mod I've posted on the forums ( http://smf.cataclysmdda.com/index.php?topic=10822.0 ), divided to avoid the PR bloat issues encountered when I first PR'd the tank mod. So this is just weapons, and the content that supports them.

Material Addition:
1. Bronze. Remember ages ago when I added copper to the game? This is the original cause of my working on implementing copper as a material.

Book addition:
1. Arms and Armor of Imperial China. Because the Chinese weapons wouldn't be easily shoehorned into other weapon and armor books.

Weapons:
1. Arming swords.
2. Longsword.No longer able to switch stances.
3. Estoc. No longer a stance-shifter.
4. Scimitar.
5. Halberd.
6. Ji, a bronze Chinese halberd-like weapon.
7. Dory, the basic Greek bronze spear.
8. Glaive.
9. Naginata, complete with option to make one by reworking a katana.
10. Jian, basic Chinese straight sword.
11. Qiang, basic Chinese spear.
12. Xiphos, Greek bronze shortsword.
13. Khopesh, Egyptian bronze sword.
14. Dao, Chinese sabre. Went with the bronze version since we have craploads of steel choppy things even in vanilla.
15. The fire lance. Likely will be awesome but impractical.
16. Fake versions of arming sword, longsword, estoc, halberd, scimitar, and jian.
17. A debug-only weapon Kadian pestered me to add. <3

Feedback on stats, both to balance content and to ensure sufficient statistical differences. Suggestions for whether to mainline any additions are likewise welcome.

Barhandar commented on data/mods/Medieval_Stuff/tools.json:
```diff
@@ -0,0 +1,543 @@
+    "color": "yellow",
+    "name": "khopesh",
+    "name_plural": "khopeshes",
+    "description": "This ancient bronze weapon features a curved, sickle-like blade sharpened on the outside edge. Associated with the New Kingdom period of ancient Eygpt, it was designed mainly to hack through the light armor common to the region.",
```
>ancient Eygpt

Also, no hook attacks?

Barhandar commented:
"Bronze" and "DURABLE_MELEE" do not go well together.

chaosvolt commented:
Ah, more spelling fails on my part. And I think we do have a hook attack of some sort I can use for it. As for bronze's durability...hmm.

Barhandar commented:
>EDIT 3: Enlightening so far, will have to determine whether these weapons still warrant retaining the flag. 

I wonder if it's possible to make a certain tag only get added if you have certain level of relevant skill. And show in description of a weapon that is less/more durable than most.

Probably would result in people remaking their weapons after a while. Also, having "[DAMAGES_EASILY]" tag that makes the tool in question _have the chance to get damaged during crafting or other non-combat actions_ would be funny. And annoy players.

chaosvolt commented:
That would be interesting, though potentially complex to add. At least for someone as bad at source tweaks as I am.

On a related note, I should possibly make the recipe for bronze booklearned instead of autolearned. It's like with blackpowder, as nice as it'd be to craft it automatically, the actually ratio isn't something the average person would know about, especially since you'd want a good ratio for making durable weapons.

And finally...casting. Not sure what tools I'll need to implement bronze casting. Forging bronze can actually be done, but it is way more finicky than than working with iron.

Barhandar commented:
>On a related note, I should possibly make the recipe for bronze booklearned instead of autolearned. It's like with blackpowder, as nice as it'd be to craft it automatically, the actually ratio isn't something the average person would know about, especially since you'd want a good ratio for making durable weapons.

If you've been educated in fabrication-specific jobs, like smithing, you're pretty much guaranteed to have this. Some professions need to have recipes memorized at the start.
>And finally...casting. Not sure what tools I'll need to implement bronze casting.

There's already a catch-all crucible. And given we can melt steel easily, you could just use that - or add a "set of metal casting molds" item and require it in recipes where appropriate.

chaosvolt commented:
Ah, so I guess the crucible is adequate. Molds however are another thing entirely. I'd assumed the crucible was to heat metal up to a workable temperature, not all the way to the point of melting it.

Either way...hmm. We could add recipes for lost-wax casting, but disposable molds would get extremely annoying.

kevingranade commented:
Interesting possibility for casting would be to make a non-reusable mold, e.g. you have one recipe to make the mold with wax and clay, and then apply a metalworking recipe to do the pour and cast your item.

chaosvolt commented:
> We could add recipes for lost-wax casting, but disposable molds would get extremely annoying.

That in a nutshell, basically. It would be a tad annoying to the player though. Though it you used a method that made the wax and plaster recoverable, it would be slightly less annoying to players.

kevingranade commented:
It's not like you're going to be casting a million of something :)

chaosvolt commented:
True, is it something I should add then? I'd be more in favor of dedicated molds, or making some of the mold resources reusable (like I did for the casting lead sling bullets recipe).

What about whether to make bronze booklearn-only?

kevingranade commented:
Sorry if I gave a bad impression, if you want to do it that'd be cool, and
I'd like it to be in eventually, but not a must for this PR.

chaosvolt commented:
Hmm. It's quite fine, I'm all for the idea of bronze casting, I'm just unsure of the best way to implement it without causing it to be too unwieldly to do. ^^"

Barhandar commented:
>Either way...hmm. We could add recipes for lost-wax casting, but disposable molds would get extremely annoying.

>True, is it something I should add then? I'd be more in favor of dedicated molds

Take 2 "wax" to make mold, have mold-using recipes also return 1 "wax", or even shorten it to recipes themselves take "wax" to represent the mold and returning only part of it that character managed to "reclaim". Dedicated molds are pointless tedium. After all, for example, you're using a "hammering"-use tool for all metalworking. And "aluminium tongs" for all of it, again. [Now compare with actual metalworking shop: http://www.spaco.org/Blacksmithing/Pictures/Eurotools/HammerRackA.jpg](http://www.spaco.org/Blacksmithing/Pictures/Eurotools/HammerRackA.jpg)

As the hammers and tongs were condensed into one tool, so the molds should be. Even if it's called a "set" like swages and dice.

chaosvolt commented:
You would also need plaster, and water, so maybe keeping it abstract would be best.

Clay might be better actually, if we're keeping it abstract.

Barhandar commented:
>You would also need plaster, and water, so maybe keeping it abstract would be best.

Alternately, add gypsum plaster to both construction and medical and plaster casts which prompt the player to remove them and get destroyed on removal, to justify adding plaster in the first place.

chaosvolt commented:
Would be interesting, but if the way bronzeworking is now is deemed acceptable, then it can be saved for a later addition.

Rivet-the-Zombie commented:
I really like this stuff, but why put it in a mod? It would be good in the base game.

chaosvolt commented:
Hmm, how much of it should be mainlined then? Should the longsword and estoc remain the way they are, and what about bronze?

Coolthulhu commented on data/mods/Medieval_Stuff/recipes.json:
```diff
@@ -0,0 +1,891 @@
+    ],
+    [
+      [ "copper", 50 ],
+      [ "cu_pipe", 1 ]
```
"cu_pipe" is 2 volume. Bronze is just 1 volume. You should probably adjust the values, for example by making it work like 1 (volume) tin + 2 (volume) copper = 3 (volume) bronze.

chaosvolt commented:
Oh right. One thing to note is I added the option to use a copper pipe BEFORE I added the dismantle recipe for it. Now that a copper pipe can be broken down into 70 copper, this can be removed.

I should've already calcutated the weight of a single unit of bronze, with the required amounts of copper/tin and the amount of bronze needed for bronzesmithing should reflect this.

Then again, several of the newer bronze weapons might not have their material-to-weight values set right.

Coolthulhu commented on data/mods/Medieval_Stuff/tools.json:
```diff
@@ -0,0 +1,543 @@
+    "price": 100000,
+    "material": "steel",
+    "flags" : ["CHOP", "DURABLE_MELEE", "SHEATH_SWORD"],
+    "techniques": ["RAPID", "WBLOCK_1"],
```
Rapid on swords this big doesn't make much sense (katana shouldn't have it either). Broadsword doesn't have it and it's smaller than those.
Swords of this size are too heavy to be used for rapid strikes.

Coolthulhu commented on data/mods/Medieval_Stuff/tools.json:
```diff
@@ -0,0 +1,543 @@
+        },
+        {
+            "type": "auto_transform",
+            "msg": "You shift to a two-handed stance.",
```
This would be better as a style.

Also, losing rapid strike by adding an extra hand doesn't make any amount of sense.

Coolthulhu commented on data/mods/Medieval_Stuff/tools.json:
```diff
@@ -0,0 +1,543 @@
+    "price": 50000,
+    "material": ["wood", "steel"],
+    "flags": ["CHOP", "DURABLE_MELEE", "REACH_ATTACK"],
+    "techniques": ["WBLOCK_1", "WIDE", "IMPALE"],
```
Halberd shouldn't impale. Impaling strike refers to attacking 2 opponents in a row. The axe blade on halberd prevents that.

Coolthulhu commented on data/mods/Medieval_Stuff/tools.json:
```diff
@@ -0,0 +1,543 @@
+    "ammo": "NULL",
+    "revert_to": "null",
+    "qualities": [["AXE", 2]],
+    "use_action": "LUMBER"
```
It would be VERY unwieldy when used to chop wood.

Coolthulhu commented on data/mods/Medieval_Stuff/tools.json:
```diff
@@ -0,0 +1,543 @@
+    "price": 100000,
+    "material": "steel",
+    "flags" : ["CHOP", "DURABLE_MELEE", "SHEATH_SWORD"],
+    "techniques": ["RAPID", "PRECISE", "WBLOCK_1"],
```
Same here. Rapid for an 8 volume weapon is rather unrealistic.
Not sure about precise. I think bumping to-hit would fit it better, if it is easy to strike precisely with.

chaosvolt commented:
Rapid strike can be removed I guess, but I would like the option to use either one or two hands on these swords.

How else to implement that feature? I guess one could implement different stances as different martial arts, but I can't think of a way to bundle them so that you get both from one purchase of Melee Weapons Training.

Coolthulhu commented on data/mods/Medieval_Stuff/tools.json:
```diff
@@ -0,0 +1,543 @@
+    "volume": 7,
+    "bashing": 10,
+    "cutting": 26,
+    "flags" : ["CHOP", "DURABLE_MELEE"],
```
Katana slices, why does dao chop?

Coolthulhu commented on data/mods/Medieval_Stuff/melee.json:
```diff
@@ -0,0 +1,186 @@
+        "color" : "yellow",
+        "symbol" : "/",
+        "material" : ["bronze", "wood"],
+        "techniques" : ["WBLOCK_1", "RAPID", "IMPALE"],
```
Rapid on spear technically makes sense, but it can cause problems in game due to how it "shuts down" slow creatures easily.

chaosvolt commented:
Katana presumably chops due to deriving it from the khopesh, which was in turn derived either from the machete or kukuri. This can also be tweaked, sure.

Coolthulhu commented:
> I would like the option to use either one or two hands on these swords.

It doesn't make sense at the moment. You could re-implement it when shields exist or something like that.
The only time when it would matter would be when one of your arms is damaged, which is a rare occurrence, especially when you're so high-level that you use swords.

chaosvolt commented:
Hmm. Shields will likely be in the next PR, but they're kinda just armor at present, so even then this distinction wouldn't be important unless we could make shields affect wielded weapons.

Should I retain the one-handed version or the two-handed version? Note that, last I checked, not even the zweihander has the "ALWAYS_TWOHAND" flag.

Coolthulhu commented:
> Shields will likely be in the next PR

I meant real shields, not just an armor worn on arm. If you make your shields the latter, they will do absolutely nothing to prevent wearing 2 shields at once and still 2handing the sword.

Oh, and 1handing the sword won't save you from arm penalty of the other arm.

At the moment pretty much every weapon is hardcoded to be used with 2 hands.

chaosvolt commented:
Ah, I see. Very well then, will implement the one-handed versions?

Coolthulhu commented:
> Should I retain the one-handed version or the two-handed version?

I'd say compare the stats to broadsword. If it has higher volume and weight (and thus attack time), rescale the damage so that dpm (damage per move) is roughly the same.
You could add wide strike to it, but I'm not sure if it's worth it. Wide strike is kinda underwhelming.

Brutal strike can be achieved simply by switching style to brawling.

chaosvolt commented:
Hmm. Need to mess with and calculate how to get that similar DPM.

Coolthulhu commented:
Moves per attack: `65 + 4 * volume() + weight() / 60`

chaosvolt commented:
Ah, thank you.

chaosvolt commented:
Hmm. What occurs if you use both "CHOP" and "STAB" on one weapon? No real way to implement the fact that a halberd can both chop and use its spearhead for stabbing.

Coolthulhu commented:
Yeah, currently it isn't possible.
"STAB" makes the weapon not-cut, "CHOP" affects only stuck penalty.

chaosvolt commented:
Ah, I see. Almost all our reach weapons are stabby things though, I think...having the halberd as a chopping version and naginata for slicing seems like a way to add variety.

chaosvolt commented:
There we go. Also on the todo list...

* Checking that all bronzesmithing recipes use an amount of bronze scraps accurate to the weapon's weight, or to the assumed weight of the metal part for hafted weapons.
* Further DPM adjustments for the other weapons.
* Determine which flags go to which polearms.

chaosvolt commented:
I would still like to implement shields eventually, but I can just add armor in Series 2 and further delay adding. Encumbrance can be used to make them interfere with weapon usage, and shields include coverage of the hands so they will slow down melee attacks, wghich will be disproportionately noticeable with two-handed weapons.

Coolthulhu commented:
> shields include coverage of the hands so they will slow down melee attacks, which will be disproportionately noticeable with two-handed weapons.

Disproportionately low. Two-handed weapons have inherently high move costs (due to their bulkiness), so a flat move cost extra here doesn't change much.
Smaller, faster weapons are affected much more, because +80 moves per attack may easily double their attack time.

chaosvolt commented:
Hmm. Granted, when you put it that way it's understandable, though I would think that being slower in general is bad in absolute terms, even if it's a minor penalty in general terms.

Still, I would hope that adding the suggested flag would be the best way to make "shields as armor" an option? We'd still need to allow some way actively block with it, instead of its passive function as armor, but it's use of coverage percent and body parts covered are adequate for representing their usefulness against arrows and other ranged attacks (or lack thereof, against firearms).

Coolthulhu commented:
> We'd still need to allow some way actively block with it, instead of its passive function as armor

That would be quite a bit more complex. It would need to be a totally separate PR, as it would involve adding a new mechanic.
I've got it worked out quite well, though. The system would work like that:
* When wielding a weapon one-handed (includes fist), you're allowed an arm block regardless of style. Styles without arm blocks would get a buff here, so it could be penalized somehow
* When targeted by a melee attack, character checks own arms for which one is better to use for a block. Some simple calculation here, comparing HP and armor on each arm.
* Now, armor calculations wouldn't be just simple armor sums, but would be based on block roll, coverage etc. Block roll would extend coverage of each item on the arm, so that even 30% coverage arm guards could be used to effectively block
* Shields would have a flag that would give a buff to their coverage, but only when used to block.

chaosvolt commented:
Hmm. That could work, yeah. Basically allowing armor to factor into blocking? Combine with a way to make shields prevent two-handed wielding and that would be ideal. o.o

Zaweri commented:
RE: the shield issue, you could add bucklers. Those allow you to use a weapon in the hand.

chaosvolt commented:
Bucklers are part of the full version of the mod, yeah. If we add a "restricts two-handed use" flag then it'd likely be the only shield to lack the flag.

chaosvolt commented:
And I just remembered I was eventually planning to add fire lances to the mod. Doh. If anyone has suggestions on stats, and whether to use an existing muzzleloading ammo versus adding a new type for it...?

kevingranade commented on data/mods/Medieval_Stuff/recipes.json:
```diff
@@ -0,0 +1,890 @@
+[
+{
+        "type" : "recipe",
```
If you're going to copy-paste a recipe all over the place, for the love of $deity, fix the formatting first.

kevingranade commented on data/mods/Medieval_Stuff/recipes.json:
```diff
@@ -0,0 +1,890 @@
+  "reversible": false,
+  "autolearn": true,
+  "tools": [
+    [
```
This is much nicer if you single-line the groups with no alternatives, like:
```
  "tools": [
    [ [ "goggles_welding", -1 ] ],
    [ [ "crucible", -1 ] ],
    [
      [ "welder", 20 ],
      [ "welder_crude", 30 ],
```

kevingranade commented on data/mods/Medieval_Stuff/recipes.json:
```diff
@@ -0,0 +1,890 @@
+    {"id":"HAMMER","level":3,"amount":1}
+  ],
+  "tools": [
+      [
```
Please fix your indention, also single-line all these as above.

chaosvolt commented:
Ack. Can't recall where I copy-pasted that bronze recipe from prior to editing. Given how old some of this content is, it likely was all the way back from 0.B or 0.C stable stuff. Will fix it...

And okay, will work on single-lining those groups as well. Had briefly forgot there was a recent PR updating the desired format. @_@

chaosvolt commented:
Also, speaking of style...why does the norm for the qualities-tools transition to not use a new line, when the transition from tools to components DOES typically use a new line?

chaosvolt commented:
And thank you for the merge. ^^
